src/Adapter/Http/HttpConnector.php

```diff
@@ -144,6 +155,40 @@ private function request($method, $path, $body, array $options)
     }
 
     /**
+     * @param $method
+     * @param $path
+     * @param $body
+     * @param array $options
+     *
+     * @return PromiseInterface
+     *
+     * @throws ServerError
+     */
+    private function asyncRequest($method, $path, $body, array $options)
+    {
+        $self = $this;
+        $body = $this->serializeBody($body, $options);
+
+        return $this->httpClient->asyncRequest($method, $path, $body, $options)->then(
+            function ($responseContents) use ($self, $options) {
+                return $self->getResponse($responseContents, $options);
```

Comment:
You don't need to carry a `$self`, in PHP you can still use `$this`, there isn't strange bound contexts (by default).